### ISSUE TITLE
Split project in 'generation' and 'serialization' parts

### ISSUE THREAD

**Issue by holtkamp** [goetas/xsd2php#72]
As described in https://github.com/goetas/xsd2php/issues/68#issuecomment-82925990, a dependency on the complete ```xsd2php``` project is introduced when only the following files are needed during (de)serialization:

- Goetas\Xsd\XsdToPhp\Jms\Handler\BaseTypesHandler
- Goetas\Xsd\XsdToPhp\Jms\Handler\XmlSchemaDateHandler

A separate project like ```xsd2php-serialization-handlers``` would be nice to prevent having the complete ```xsd2php``` project and all of its dependencies running in production-like environments. 

Using the [suggest section of composer](https://getcomposer.org/doc/04-schema.md#suggest) one can then easily refer to this new ```xsd2php-serialization-handlers``` project...

@schmittjoh might also consider adopting these handlers in his https://github.com/schmittjoh/serializer project...

**Comment by goetas** [comment]
@holtkamp 
> @schmittjoh might also consider adopting these handlers in his https://github.com/schmittjoh/serializer project...

Ti thing that of out of scope for JMS serializer

**Comment by goetas** [comment]
Splitting the project into two components make sense!

**Comment by holtkamp** [comment]
hi @goetas , I encountered this issue again...

- we currently use the Jms Handlers during serialization
- therefore the ```composer.json``` requires ```goutas/xsd2php```
- ```goutas/xsd2php``` depends on [zend-code: ~2.2] (https://github.com/goetas/xsd2php/blob/5eb67848853d8e3693540e28d9a9692206b948d6/composer.json#L25)
- this conflicts with intended use of ```ocramius/proxy-manager``` which uses [zend-code: ~3.0] (https://github.com/Ocramius/ProxyManager/blob/6e15877fc03cc211ce798d497a21208a44f7b726/composer.json#L23)

Of course we can simply copy the Jms Handlers to prevent the need for having the whole ```goutas/xsd2php``` component, but a more elegant approach would be preferred :smile:

**Comment by holtkamp** [comment]
@goetas 

maybe it is an option to allow both zend-code 2.2 and 3.0:

```
zend-code: "~2.2 | ~3.0"
```

**Comment by goetas** [comment]
I'm fine with it, do you know if it is backward compatible?

**Comment by holtkamp** [comment]
will have a look this week. When glancing at http://framework.zend.com/blog/2016-01-14-zf3-biweekly-update.html, I expect only specifics for the new versions are added...

**Comment by holtkamp** [comment]
I noticed you have been working on https://github.com/goetas-webservices/xsd2php-runtime, nice job! Some context: this way the ```xsd2php``` project can be left out of the project that uses the generated PHP classes and YAML classes used by serializer. 

The generation of these files is done in separate projects, which only require ```xsd2php``` during development. I think this issue can be closed?

**Comment by goetas** [comment]
will close it soon once all the new stuff is merged

**Comment by goetas** [comment]
implemented in https://github.com/goetas/xsd2php/pull/108
